controllers/storageclient_controller.go

```diff
@@ -549,6 +572,56 @@ func (r *StorageClientReconciler) reconcileSharedfileStorageClaim() error {
 	return nil
 }
 
+func (r *StorageClientReconciler) reconcileRemoteNoobaa() error {
+	// Create noobaa CR referencing the join secret
+	nb := &nbv1.NooBaa{}
+	nb.Name = "noobaa-remote"
+	nb.Namespace = r.OperatorNamespace
+	nb.Spec.JoinSecret = &corev1.SecretReference{
+		Name:      joinSecretName,
+		Namespace: r.OperatorNamespace,
+	}
+	_, err := controllerutil.CreateOrUpdate(r.ctx, r.Client, nb, func() error {
+		utils.AddAnnotation(nb, "remote-client-noobaa", "true")
+		if err := r.own(nb); err != nil {
+			return err
+		}
```

Comment:
please set the spec after owning the secret. it should be inside CreateOrUpdate something like this.


```
_, err := controllerutil.CreateOrUpdate(r.ctx, r.Client, nb, func() error {
		utils.AddAnnotation(nb, "remote-client-noobaa", "true")
		if err := r.own(nb); err != nil {
			return err
		}
nb.Spec.JoinSecret = &corev1.SecretReference{
		Name:      joinSecretName,
		Namespace: r.OperatorNamespace,
	}
return nil
	})
```

Reply:
Updated